FlowCal/io.py

```diff
@@ -1760,6 +1760,38 @@ def __array_finalize__(self, obj):
         if hasattr(obj, '_resolution'):
             self._resolution = copy.deepcopy(obj._resolution)
 
+    # override __reduce__ and __setstate__ to fix Issue #277
+    # Adapted from https://stackoverflow.com/a/26599346
+    def __reduce__(self):
+        pickled_state = super(FCSData, self).__reduce__() # parent's __reduce__ output
+        # Update the __setstate__ tuple with FCSData attrs
+        fcsdata_attrs = (self._infile, self._text, self._analysis,
+            self._data_type, self._time_step, self._acquisition_start_time,
+            self._acquisition_end_time, self._channels,
+            self._amplification_type, self._detector_voltage,
+            self._amplifier_gain, self._range, self._resolution,)
+        new_state = pickled_state[2] + fcsdata_attrs
+        # Return updated tuple with full set_state tuple
+        return (pickled_state[0], pickled_state[1], new_state)
+
+    def __setstate__(self, state):
```

Comment:
We need a docstring here explaining what this function does and why it was replaced. It doesn't have to be too detailed, as this function will not be exposed to the user, but it should be enough to be useful for anyone in the future modifying this code. Please refer to https://github.com/numpy/numpy/blob/master/doc/example.py.